Src/FluentAssertions/Numeric/NumericAssertions.cs

```diff
@@ -14,21 +14,21 @@ namespace FluentAssertions.Numeric
     /// </summary>
     [DebuggerNonUserCode]
     public class NumericAssertions<T>
-        where T : struct
+        where T : struct, IComparable
     {
-        public NumericAssertions(object value)
+        public NumericAssertions(T value)
+            : this((T?)value)
         {
-            if (!(value is null))
-            {
-                Subject = value as IComparable;
-                if (Subject is null)
-                {
-                    throw new InvalidOperationException("This class only supports types implementing IComparable.");
-                }
-            }
         }
 
-        public IComparable Subject { get; }
+        private protected NumericAssertions(T? value)
```

Comment:
Why `private protected`?
Do you want forbid extensions using this class?

Reply:
I don't think I'm forbidding extensions on this class.
Did you see the other constructors?
There are two constructor available, which I think should be covering
```c#
public NumericAssertions(T value) { }
public NullableNumericAssertions(T? value) { }
```